frontend/src/App.tsx

```diff
@@ -1,39 +1,72 @@
-import { useState } from 'react';
-import logo from './logo.svg';
+import { Box, Container, Tab, TabList, TabPanel, TabPanels, Tabs } from '@chakra-ui/react';
+import { ReactNode } from 'react';
 import './App.css';
+import useConfigContext from './context/ConfigContext';
+import AllFloorsConfig from './pages/AllFloorsConfig';
+import FloorConfig from './pages/FloorConfig';
+import GeneralConfig from './pages/GeneralConfig';
+import StartingConfig from './pages/StartingConfig';
+import { Floor, Options } from './types/Configuration';
+import TabData from './types/TabData';
+import { optionsHasFlag } from './utility/ConfigHelpers';
 
 function App() {
-    const [count, setCount] = useState(0);
+    const [config, setConfig] = useConfigContext();
+
+    function getTabs(): TabData[] {
+        const tabs: TabData[] = [
+            {
+                name: 'General Config',
+                page: <GeneralConfig />,
+            },
+            {
+                name: 'StartingConfig',
+                page: <StartingConfig />,
+            },
+        ];
+
+        // TODO: Try cache components
```

Comment:
Might be the source of the performance issues?

Reply:
I've added caching using React's built-in `useCallback` method which only re-renders the tabs when it has to in #22. I think the main performance issue might actually be that (from what I can tell) Chakra's built-in tab system renders all the tabs on top of each other, but just makes them invisible. This means that changing certain options causes re-renders across all the tabs despite these changes not being visible (For example enabling weighted skill spawns causes all the skills in all the dropzones to be re-rendered). I do need to double-check this though.